src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -365,7 +365,8 @@ unsigned HWIntrinsicInfo::lookupSimdSize(Compiler* comp, NamedIntrinsic id, CORI
     {
         typeHnd = comp->info.compCompHnd->getArgClass(sig, sig->args);
     }
-    else if (HWIntrinsicInfo::BaseTypeFromSecondArg(id))
+    else if ((HWIntrinsicInfo::lookupCategory(id) == HW_Category_MemoryStore) ||
```

Comment:
Why does this need to be a secondary check, rather than just annotating the methods with `baseTypeFromSecondArg`?

Reply:
Agree, I should've annotated with the flag - I will update and also change `Compiler::getBaseTypeFromArgIfNeeded` where we do the same check

Reply:
@tannergooding Do you have any concerns with annotating most of the Store-methods in hwintrinsiclistxarch.h with HW_Flag_BaseTypeFromSecondArg?

Reply:
I don't believe so. I might be forgetting something that prevented this, but nothing immediately comes to mind.

Reply:
> 
> 
> I don't believe so. I might be forgetting something that prevented this, but nothing immediately comes to mind.

Let's see whether the x86 tests are going to pass with this change

Reply:
Addressed this one.